### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by julianrod94**
Changes to the Application:
- You will be able to choose the type of connection before the compilation of the code. You wont be able to change it unless you compile it again.
- New "Request" object is needed because the user should not be aware of the type of connection used. 
- New folders for namedPipe and socket implementation.
- All Request and Response data are now handled as pointers.
- Server Makefile working

### CODE REVIEW

File: Api/namedPipe/namedPipe.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#include <sys/types.h>
+#include <sys/time.h>
+#include <unistd.h>
+
+
+#define BLOCK 1024 
+
+int openNamedPipe(char * something);
+
+void writeNamedPipe(int fd, void * data, int size);
```
Comment (jjsebasv) [inline Api/namedPipe/namedPipe.h:13]: Didn't he tell us that `fd` shouln't be shown here. Perhaps we could change its name

**Comment by jjsebasv** [comment]
Fantastic :+1: 
I'm Merging this

File: Api/request.c [hunk 2]
```diff
@@ -1,42 +1,38 @@
...
+  createResponse(response,fd[0]);
+  if(response == NULL){
+    return NULL;
   }
+  return SUCCESS; 
+}
 
-  return REQUEST_OK;
+//TODO
+int getReponse(Response * response){
```
Comment (jjsebasv) [inline Api/request.c:30]: I don't get this

Comment (julianrod94) [inline Api/request.c:30]: you provide the direction (heep) where the response will be saved.

Comment (jjsebasv) [inline Api/request.c:30]: Isn't this function getting a response?

File: Api/namedPipe/namedPipe.h (hunk 1 above)
Comment (julianrod94) [inline Api/namedPipe/namedPipe.h:13]: Those should be "private" functions( I don't know how to do it in C. Java <3.), there should be one function (the one on the user's side) that call this one.

**Pull request merged**